beam_jobs/base_model_validator.py

```diff
@@ -0,0 +1,239 @@
+import python_utils
+
+
+
+(base_models, user_models) = models.Registry.import_models(
+    [models.NAMES.base_model, models.NAMES.user])
+datastore_services = models.Registry.import_datastore_services()
+
+
+ERROR_CATEGORY_COMMIT_CMD_CHECK = 'commit cmd check'
+ERROR_CATEGORY_COMMIT_STATUS_CHECK = 'post commit status check'
+ERROR_CATEGORY_COUNT_CHECK = 'count check'
+ERROR_CATEGORY_CURRENT_TIME_CHECK = 'current time check'
+ERROR_CATEGORY_DATETIME_CHECK = 'datetime check'
+ERROR_CATEGORY_DOMAIN_OBJECT_CHECK = 'domain object check'
+ERROR_CATEGORY_EMAIL_CHECK = 'email check'
+ERROR_CATEGORY_ERROR_CHECK = 'error check'
+ERROR_CATEGORY_FIELD_CHECK = 'field check'
+ERROR_CATEGORY_FIRST_PUBLISHED_MSEC_CHECK = 'first published msec check'
+ERROR_CATEGORY_ID_CHECK = 'id check'
+ERROR_CATEGORY_LAST_UPDATED_CHECK = 'last updated check'
+ERROR_CATEGORY_LENGTH_CHECK = 'length check'
+ERROR_CATEGORY_NAME_CHECK = 'name check'
+ERROR_CATEGORY_OUTPUT_CHECK = 'output check'
+ERROR_CATEGORY_PRIVATE_COMMIT_CHECK = 'post commit is private check'
+ERROR_CATEGORY_PROPERTY_FETCH_CHECK = 'fetch properties'
+ERROR_CATEGORY_RATED_ON_CHECK = 'rated on check'
+ERROR_CATEGORY_RATINGS_CHECK = 'ratings check'
+ERROR_CATEGORY_REFERENCE_CHECK = 'reference check'
+ERROR_CATEGORY_AUTHOR_CHECK = 'author check'
+ERROR_CATEGORY_REVIEWER_CHECK = 'reviewer check'
+ERROR_CATEGORY_STATE_NAME_CHECK = 'state name check'
+ERROR_CATEGORY_SUMMARY_CHECK = 'summary check'
+ERROR_CATEGORY_TIME_FIELD_CHECK = 'time field relation check'
+ERROR_CATEGORY_TYPE_CHECK = 'type check'
+ERROR_CATEGORY_VERSION_CHECK = 'version check'
+ERROR_CATEGORY_STALE_CHECK = 'stale check'
+ERROR_CATEGORY_INVALID_IDS_IN_FIELD = 'invalid ids in field'
+
+VALIDATION_MODE_NEUTRAL = 'neutral'
+VALIDATION_MODE_STRICT = 'strict'
+VALIDATION_MODE_NON_STRICT = 'non-strict'
+
+
+class ValidateModelIdWithRegex(beam.DoFn):
+    """DoFn to validate model ids against a given regex string."""
+
+    def __init__(self, regex_string):
+        """Initializes the ValidateModelIdWithRegex DoFn.
+
+        Args:
+          regex_string: str. A regex pattern for valid ids.
+        """
+        super(init)
+        self.regex_string = regex_string
+
+    def process(self, model):
+        """Function that defines how to process each element in a pipeline of
+        models.
+
+        Args:
+          model: datastore_services.Model. Entity to validate.
+
+        Yields:
+          beam.pvalue.TaggedOutput. An element of the output PCollection for
+          the doFn which represents an error as a key value pair.
+        """
+        element = model.clone()
+        regex_string = self.regex_string
+
+        if not re.compile(regex_string).match(element.id):
+            yield beam.pvalue.TaggedOutput('error_category_id_check', (
+                'model %s' % ERROR_CATEGORY_ID_CHECK,
+                'Entity id %s: Entity id does not match regex pattern' % (
+                    element.id)
+            ))
+        yield element
```

Comment:
Can we just delete this? How do we expect these elements to be used?

Reply:
I think we can delete this. I had this in there at first just in case we'd need the elements yielded along with the errors.